packages/build-utils/test/unit.builds-and-routes-detector.test.ts

```diff
@@ -78,18 +78,62 @@ describe('Test `detectBuilders`', () => {
     expect(builders![0].use).toBe('@vercel/node');
     expect(builders![0].src).toBe('api/users.js');
     expect(builders![1].use).toBe('@vercel/static');
-    expect(builders![1].src).toBe('!{api/**,package.json}');
+    expect(builders![1].src).toBe('!{api/**,package.json,middleware.[jt]s}');
     expect(builders!.length).toBe(2);
     expect(errors).toBe(null);
   });
 
+  it('no package.json + no build + root-level "middleware.js"', async () => {
+    const files = ['middleware.js', 'index.html', 'web/middleware.js'];
+    const { builders, errors } = await detectBuilders(files);
+    expect(builders![0].use).toBe('@vercel/node');
+    expect(builders![0].src).toBe('middleware.js');
+    expect(builders![0].config?.middleware).toEqual(true);
+    expect(builders![1].use).toBe('@vercel/static');
+    expect(builders![1].src).toBe('!{api/**,package.json,middleware.[jt]s}');
+    expect(builders!.length).toBe(2);
+    expect(errors).toBe(null);
+  });
+
+  it('no package.json + no build + root-level "middleware.ts"', async () => {
+    const files = ['middleware.ts', 'index.html', 'web/middleware.js'];
+    const { builders, errors } = await detectBuilders(files);
+    expect(builders![0].use).toBe('@vercel/node');
+    expect(builders![0].src).toBe('middleware.ts');
+    expect(builders![0].config?.middleware).toEqual(true);
+    expect(builders![1].use).toBe('@vercel/static');
+    expect(builders![1].src).toBe('!{api/**,package.json,middleware.[jt]s}');
+    expect(builders!.length).toBe(2);
+    expect(errors).toBe(null);
+  });
+
+  it('should not add middleware builder when "nextjs" framework is selected', async () => {
```

Comment:
Lets move all these new tests into the section that uses `const featHandleMiss = true;`

In the future we need delete the other code path because that is no longer used (featHandleMiss is always true)

Reply:
df6d714c1d4d36951d1efea2af88898eba02ec85